desc/compute/_neoclassical.py

```diff
@@ -0,0 +1,275 @@
+    resolution_requirement="z",  # and poloidal if near rational surfaces
+    source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
+)
+def _L_ra_fsa(data, transforms, profiles, **kwargs):
+    g = transforms["grid"].source_grid
+    shape = (g.num_rho, g.num_alpha, g.num_zeta)
+    L_ra = simpson(
+        y=jnp.reciprocal(data["B^zeta"]).reshape(shape),
+        x=jnp.reshape(g.nodes[:, 2], shape),
+        axis=-1,
+    )
+    data["<L|r,a>"] = g.expand(jnp.abs(_poloidal_mean(g, L_ra)))
+    return data
+
+
+@register_compute_fun(
+    name="<G|r,a>",
+    label="\\int_{\\zeta_{\\mathrm{min}}}^{\\zeta_{\\mathrm{max}}}"
+    " \\frac{d\\zeta}{|B^{\\zeta} \\sqrt g|}",
+    units="1 / Wb",
+    units_long="Inverse webers",
+    description="(Mean) proper length over volume of field line(s)",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="r",
+    data=["B^zeta", "sqrt(g)"],
+    resolution_requirement="z",  # and poloidal if near rational surfaces
+    source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
+)
+def _G_ra_fsa(data, transforms, profiles, **kwargs):
+    g = transforms["grid"].source_grid
+    shape = (g.num_rho, g.num_alpha, g.num_zeta)
+    G_ra = simpson(
+        y=jnp.reciprocal(data["B^zeta"] * data["sqrt(g)"]).reshape(shape),
+        x=jnp.reshape(g.nodes[:, 2], shape),
+        axis=-1,
+    )
+    data["<G|r,a>"] = g.expand(jnp.abs(_poloidal_mean(g, G_ra)))
+    return data
+
+
+@register_compute_fun(
+    name="effective ripple",  # this is ε¹ᐧ⁵
+    label=(
+        # ε¹ᐧ⁵ = π/(8√2) (R₀/〈|∇ψ|〉)² ∫dλ λ⁻²B₀⁻¹ 〈 ∑ⱼ Hⱼ²/Iⱼ 〉
+        "\\epsilon^{3/2} = \\frac{\\pi}{8 \\sqrt{2}} "
+        "(R_0 / \\langle \\vert\\nabla \\psi\\vert \\rangle)^2 "
+        "\\int d\\lambda \\lambda^{-2} B_0^{-1} "
+        "\\langle \\sum_j H_j^2 / I_j \\rangle"
+    ),
+    units="~",
+    units_long="None",
+    description="Effective ripple modulation amplitude",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="r",
+    data=[
+        "min_tz |B|",
+        "max_tz |B|",
+        "B^zeta",
+        "|B|",
+        "|B|_z|r,a",
+        "|grad(rho)|",
+        "kappa_g",
+        "<L|r,a>",
+        "R0",
+        "<|grad(rho)|>",
+    ],
+    resolution_requirement="z",  # and poloidal if near rational surfaces
+    source_grid_requirement={"coordinates": "raz", "is_meshgrid": True},
+    num_quad="int : Resolution for quadrature of bounce integrals. Default is 31.",
+    num_pitch=(
+        "int : Resolution for quadrature over velocity coordinate, preferably odd. "
+        "Default is 125. Effective ripple will look smoother at high values. "
+        "(If computed on many flux surfaces and micro oscillation is seen "
+        "between neighboring surfaces, increasing num_pitch will smooth the profile)."
+    ),
+    num_wells=(
+        "int : Maximum number of wells to detect for each pitch and field line. "
```

Comment:
is this the total number of wells across the entire length of the field line, per field line per pitch?

i.e.if I say 10, then only 10 wells will be detected per field line and per pitch angle value?